app/src/main/java/org/gdg/frisbee/android/pulse/PulseAdapter.java

```diff
@@ -117,23 +122,16 @@ public void setPulse(final int mode, Pulse pulse) {
         mMode = mode;
         mPulse.clear();
         mPulse.addAll(pulse.entrySet());
-        mPosition = new int[mPulse.size()];
+        //Only initialize if the positions are not provided.
+        //They are provided in constructor and coming from savedInstanceState of the Fragment.
+        if (mPositions == null || mPositions.length != mPulse.size()) {
+            mPositions = new int[mPulse.size()];
+        }
 
         Collections.sort(mPulse, new Comparator<Map.Entry<String, PulseEntry>>() {
             @Override
             public int compare(Map.Entry<String, PulseEntry> entry, Map.Entry<String, PulseEntry> entry2) {
-                PulseEntry value = entry.getValue();
-                PulseEntry value2 = entry2.getValue();
-
-                switch (mode) {
-                    case 0:
-                        return (value.getMeetings() - value2.getMeetings()) * -1;
-                    case 1:
-                        return (value.getAttendees() - value2.getAttendees()) * -1;
-                    case 2:
-                        return (value.getPlusMembers() - value2.getPlusMembers()) * -1;
-                }
-                return 0;
+                return entry.getValue().compareTo(mode, entry2.getValue());
```

Comment:
The type of the objects is `Map.Entry<String, PulseEntry>` not just `PulseEntry`. Map.Entry does not implement Comparable. So it will not work. Getting rid of `Map.Entry<String, PulseEntry>` is out of this PR's scope I think.